pkg/controller/acmechallenges/sync_test.go

```diff
@@ -110,14 +110,6 @@ func TestSyncHappyPath(t *testing.T) {
 					testIssuerHTTP01Enabled,
 				},
 				ExpectedActions: []testpkg.Action{
-					testpkg.NewAction(coretesting.NewUpdateSubresourceAction(cmacme.SchemeGroupVersion.WithResource("challenges"),
-						"status",
-						gen.DefaultTestNamespace,
-						gen.ChallengeFrom(deletedChallenge,
-							gen.SetChallengeProcessing(true),
-							gen.SetChallengeURL("testurl"),
-							gen.SetChallengeType(cmacme.ACMEChallengeTypeHTTP01),
-						))),
```

Comment:
This change reflects the fact that the original implementation unconditionally called UpdateStatus, regardless of whether any of the status fields had been changed....which they are not during cleanup

Reply:
I guess we could though have a test case where a challenge is deleted, but the clean up fails. In that case we would expect that the `Challenge` Reason is set to whatever the clean up error was (which is probably pretty useless as the `Challenge` will get garbage collected) and that an error event is thrown.

Reply:
Good idea. I've added that test.

Reply:
I found out that until recently the documentation on this was incorrect:
 * https://github.com/kubernetes/kubernetes/issues/108025

And there is still an incorrect paragraph in https://kubernetes.io/blog/2021/05/14/using-finalizers-to-control-deletion/
> In short, what’s happened is that the object was updated, not deleted. That’s because Kubernetes saw that the object contained finalizers and put it into a read-only state. The deletion timestamp signals that the object can only be read, with the exception of removing the finalizer key updates.